test/gtest/soft_max_new.cpp

```diff
@@ -0,0 +1,161 @@
+#include <miopen/miopen.h>
+#include <miopen/softmax.hpp>
+#include <miopen/tensor.hpp>
+#include <utility>
+#include <algorithm>
+
+#include "driver.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+
+#include <cstdint>
+#include <tuple>
+#include <gtest/gtest.h>
+
+struct InputDimension
+{
+    size_t N;
+    size_t C;
+    size_t H;
+    size_t W;
+
+    friend std::ostream& operator<<(std::ostream& os, const InputDimension& input)
+    {
+        return os << "(N: " << input.N << " C:" << input.C << " H:" << input.H << " W:" << input.W
+                  << ")";
+    }
+};
+// this is just try
+std::set<std::vector<int>> GetInputDimensions()
+{
+    return {
+        {1, 2, 3, 4},
+        {4, 3, 2, 1},
+    };
+}
+
+struct Scales
+{
+    float alpha;
+    float beta;
+
+    friend std::ostream& operator<<(std::ostream& os, const Scales& ab)
+    {
+        return os << "(alpha: " << ab.alpha << " beta:" << ab.beta << ")";
+    }
+};
+
+template <typename DataType>
+class SoftMaxTest
+    : public ::testing::TestWithParam<
+          std::tuple<InputDimension, miopenSoftmaxAlgorithm_t, miopenSoftmaxMode_t, Scales>>
+{
+public:
+    static std::vector<InputDimension> convertDim(const std::set<std::vector<int>>& input)
+    {
+
+        std::vector<InputDimension> result;
+        result.reserve(input.size());
+        std::transform(input.begin(),
+                       input.end(),
+                       std::back_inserter(result),
+                       [](const std::vector<int>& inp) {
+                           return InputDimension{static_cast<size_t>(inp[0]),
+                                                 static_cast<size_t>(inp[1]),
+                                                 static_cast<size_t>(inp[2]),
+                                                 static_cast<size_t>(inp[3])};
+                       });
+        return result;
+    }
+
+protected:
+    void SetUp() override
+    {
+        std::tie(input, algo, mode, scales) = GetParam();
+
+        auto&& handle = get_handle();
+
+        // initialze output tensor with same dim as first input tensor
+        output = tensor<DataType>{out_dim};
+        std::fill(output.begin(), output.end(), std::numeric_limits<DataType>::quiet_NaN());
+
+        auto output_dev = handle.Write(output.data);
+    }
+    void Print()
+    {
+        // print input test parameters
+        std::cout << "Run test:" << std::endl;
+        std::cout << "Input Dimensions: " << input << std::endl;
+        std::cout << "Softmax Algorithm: " << algo << std::endl;
+        std::cout << "Softmax Mode: " << mode << std::endl;
+        std::cout << "Scales: " << scales << std::endl;
+    }
+
+    void TearDown() override
+    {
+        // probably deallocate anything if required
+    }
+
+private:
+    // keep all the input\output intermediate data here
+
+    InputDimension input;
+    tensor<DataType> output;
+    tensor<DataType> out_dim;
```

Comment:
I guess `input` should be a `tensor<>`, `out_dim` should be removed and replaced with `InputDimension input_dim`.